text/0074-global-init.md

```diff
@@ -0,0 +1,176 @@
+// Meter provider acts as a "forwarding" SDK.  The default global
+// Meter provider will begin forwarding to the installed Meter
+// provider once this is called.
+func SetMeterProvider(metric.Provider) { ... }
+
+// SetPropagators initializes the global Propagators instance.  May only
+// be called once per process lifetime.  Subsequent calls will panic.
+//
+// Prior to setting the global Meter provider, the default global
+// Propagators instance performs pass-through W3C Traceparent and
+// Correlation-Context propagation.
+func SetPropagators(propagation.Propagators) { ... }
+
+// SetPropagators initializes the global Tracer provider.  May only
+// be called once per process lifetime.  Subsequent calls will panic.
+//
+// Prior to setting the global Tracer provider, the default global
+// Tracer provider acts as a "forwarding" SDK.  The default global
+// Tracer provider will begin forwarding to the installed Tracer
+// provider once this is called.
+func SetTraceProvider(trace.Provider) { ... }
+```
+
+#### Requirements
+
+We anticipate third party libraries using the global instances before
+they are installed, and we wish for references obtained through these
+instances to become functional once the corresponding implementation
+is initialized.  The default instances returned by the global getters
+for Tracer provider, Meter provider, and Propagators must "forward" to
+the real SDK implementation once it is installed.
+
+#### Tracer
+
+Tracers obtained through the provider will become functional when the
+user's Tracer SDK is installed as the global instance.
+
+Spans started prior to installing the Tracer SDK will be No-op spans.
+Installing a Tracer SDK after starting a span via the default global
+instance does not change this behavior.
+
+#### Meter
+
+Meters obtained through the provider will become functional when the
+users's Meter SDK is installed as the global instance.
+
+Metric events will be dropped until the Meter SDK is installed.
+
+#### Propagators
+
+The default global Propagators instance will by default perform
+pass-through W3C Traceparent and Correlation-Context propagation.
+
+The default global Propagators instance will begin forwarding to the
+user's Propagators when it is installed as the global instance.
+
+## Trade-offs and mitigations
+
+### Testing support
+
+Testing should be performed without depending on the global SDK, if
+possible.  A convenience method may be provided for tests to reset the
+global state to the initial, default conditions.
+
+### Efficiency concern
+
+Since the global instances are required to begin working once the real
+implementations are installed, there is some implied synchronization
+overhead and cost.  This overhead SHOULD be minimal.
+
+We recommend to explicitly install a No-op instance to lower the cost
+of instrumentation when no SDK will be installed, as opposed to
+leaving the default global instances in place, perpetually waiting to
```

Comment:
I don't follow this - what is the difference between "installed Noop instance" and "default global instance"?

Reply:
I think "default" here is referring to the default SDK implementation that ships with Open Telemetry.

Reply:
In other words, a user needs to do _something_ to install the default SDK? Is that really wise, given that in most cases people _will_ want a real SDK? Perhaps the Noop one should be an option to install.

Reply:
I'm not sure I understand what "most cases" means. If I'm adding instrumentation to a library, I only have access to the API, and users of my library should definitely get the No-op version if they haven't wired up an SDK in some way.

Reply:
I think I should remove this paragraph. I'm saying that it will be somewhat more expensive to use the global forwarding SDK but never install an SDK, than to use an explicitly installed No-op SDK.